pennylane/pauli/conversion.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Utility functions to convert between ``~.PauliSentence`` and other PennyLane operators.
+"""
+from operator import matmul
+from itertools import product
+from typing import Union
+from functools import reduce, singledispatch
+
+import numpy as np
+
+from .pauli_arithmetic import PauliWord, PauliSentence, I, X, Y, Z, mat_map, op_map, op_to_str_map
+from .utils import is_pauli_word
+
+from pennylane.ops import Sum, Prod, SProd, PauliX, PauliY, PauliZ, Identity
+from pennylane.operation import Tensor
+from pennylane.ops.qubit.hamiltonian import Hamiltonian
+
+
+@singledispatch
+def pauli_sentence(op):
+    """Return the PauliSentence representation of an arithmetic operator or Hamiltonian.
+
+    Args:
+        op (~.Operator): The operator or hamiltonian that needs to converted.
+
+    Raises:
+        ValueError: Op must be a linear combination of Pauli operators
+
+    Returns ~.PauliSentence
+    """
+    raise ValueError(f"Op must be a linear combination of Pauli operators only, got: {op}")
+
+
+@pauli_sentence.register
+def _(op: PauliX):
+    return PauliSentence({PauliWord({op.wires[0]: X}): 1.0})
+
+
+@pauli_sentence.register
+def _(op: PauliY):
+    return PauliSentence({PauliWord({op.wires[0]: Y}): 1.0})
+
+
+@pauli_sentence.register
+def _(op: PauliZ):
+    return PauliSentence({PauliWord({op.wires[0]: Z}): 1.0})
+
+
+@pauli_sentence.register
+def _(op: Identity):
+    return PauliSentence({PauliWord({}): 1.0})
+
+
+@pauli_sentence.register
+def _(op: Tensor):
+    if not is_pauli_word(op):
+        raise ValueError(f"Op must be a linear combination of Pauli operators only, got: {op}")
+
+    pw = {}
+    for factor in op.obs:
+        wire, pauli = (factor.wires[0], op_to_str_map[factor.__class__])
+        pw[wire] = pauli
+
+    return PauliSentence({PauliWord(pw): 1.0})
+
+
+@pauli_sentence.register
+def _(op: Prod):
+    factors = [pauli_sentence(factor) for factor in op.operands]
+    return reduce(lambda a, b: a * b, factors)
+
+
+@pauli_sentence.register
+def _(op: SProd):
+    ps = pauli_sentence(op.base)
+    for pw, coeff in ps.items():
+        ps[pw] = coeff * op.scalar
+    return ps
+
+
+@pauli_sentence.register
+def _(op: Hamiltonian):
+    if not is_pauli_word(op):
+        raise ValueError(f"Op must be a linear combination of Pauli operators only, got: {op}")
```

Comment:
```suggestion
```

Hamiltonian's are not going to be pauli words most of the time.

```
>>> qml.pauli.is_pauli_word(qml.PauliX(0) + qml.PauliY(1))
False
```

Reply:
That's weird, it returns true for me: 

![image](https://user-images.githubusercontent.com/21964864/199823496-4fc023a6-9361-4645-9306-3f91c5362121.png)

Reply:
I might have addressed it in a recent fix